Assets/Scripts/TES/MorrowindEngine.cs

```diff
@@ -285,33 +286,33 @@ public void Update()
 		}
 		public void CastInteractRay()
 		{
-			interactTextObj.SetActive(false);
-
 			// Cast a ray to see what the camera is looking at.
-			var ray = new Ray(playerCameraObj.transform.position, playerCameraObj.transform.forward);
+			Ray ray = new Ray(playerCameraObj.transform.position, playerCameraObj.transform.forward);
 
-			var raycastHitCount = Physics.RaycastNonAlloc(ray, interactRaycastHitBuffer, maxInteractDistance);
+			int raycastHitCount = Physics.RaycastNonAlloc(ray, interactRaycastHitBuffer, maxInteractDistance);
 
 			for(int i = 0; i < raycastHitCount; i++)
 			{
-				var hitInfo = interactRaycastHitBuffer[i];
+				RaycastHit hitInfo = interactRaycastHitBuffer[i];
 
 				// Find the door associated with the hit collider.
 				GameObject doorObj = GameObjectUtils.FindObjectWithTagUpHeirarchy(hitInfo.collider.gameObject, "Door");
 
 				if(doorObj != null)
 				{
-					var doorComponent = doorObj.GetComponent<DoorComponent>();
+					DoorComponent doorComponent = doorObj.GetComponent<DoorComponent>();
 
-					interactTextObj.SetActive(true);
+					if ( !interactTextObj.activeSelf ) interactTextObj.SetActive( true );
```

Comment:
Try to stay consistent in coding style with the rest of the code. I don't put spaces around parenthesis, just another minor style choise. Once again, it's not a huge deal, I'll merge it, but I'm going to try to clean up a lot of code soon and I might change it back. It'll just become a mess if different contributors use different style conventions everywhere.